From: Thomas Bogendoerfer <tbogendoerfer@suse.de>
Subject: kabi protect struct iw_cm_id
Patch-mainline: never, kabi
References: bsc#1136348 jsc#SLE-4684

Upstream commit 926ba19b3574 ("RDMA/iwcm: add tos_set bool to iw_cm struct")
adds a new field tos_set and modifies width of mapped in struct iw_cm_id,
which break kABI. Add kABI markers for tos_set and revert change
of mapped field.

Signed-off-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 include/rdma/iw_cm.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/include/rdma/iw_cm.h
+++ b/include/rdma/iw_cm.h
@@ -94,8 +94,10 @@ struct iw_cm_id {
 	void (*add_ref)(struct iw_cm_id *);
 	void (*rem_ref)(struct iw_cm_id *);
 	u8  tos;
+	bool mapped;
+#ifndef __GENKSYMS__
 	bool tos_set:1;
-	bool mapped:1;
+#endif
 };
 
 struct iw_cm_conn_param {
